docs/control-structures.rst

```diff
@@ -51,40 +56,45 @@ write::
 
 The names of output parameters can be omitted.
 The output values can also be specified using ``return`` statements.
-The ``return`` statements are also capable of returning multiple
-values, see :ref:`multi-return`.
+
+The ``return`` statements are also capable of returning multiple values using ``return (v0, v1, ..., vn)``. The number of parameters must equal the number of output parameters.
+
 Return parameters are initialized to zero; if they are not explicitly
 set, they stay to be zero.
 
 Input parameters and output parameters can be used as expressions in
 the function body.  There, they are also usable in the left-hand side
 of assignment.
 
+...note::
+    You can return all types from Solidity functions, but for structs, only from ``internal`` function calls. You can return ``memory`` structs from any function if you enable the new experimental ``ABIEncoderV2`` feature by adding ``pragma experimental ABIEncoderV2;`` to your source file. This feature introduces other experimental changes that you can read more about in the ABI_ guide.
+
 .. index:: if, else, while, do/while, for, break, continue, return, switch, goto
 
 Control Structures
 ===================
 
-Most of the control structures from JavaScript are available in Solidity
-except for ``switch`` and ``goto``. So
-there is: ``if``, ``else``, ``while``, ``do``, ``for``, ``break``, ``continue``, ``return``, ``? :``, with
-the usual semantics known from C or JavaScript.
+Most of the control structures of curly-bracket languages such as JavaScript or C are available in Solidity except for ``switch`` and ``goto``. The full list is:
 
-Parentheses can *not* be omitted for conditionals, but curly brances can be omitted
-around single-statement bodies.
+- ``if (<condition>) {<true-statement>}``
+- ``if (<condition>) {<true-statement>} else {<false-statement>}``
+- ``while (<condition-true>) {<true-statement>}``
+- ``do {<true-statement>} (while <condition-true>)``
```

Comment:
Perhaps better to just call this condition instead of condition-true (also in other places)?